ui/src/app/views/workspace-edit.tsx

```diff
@@ -381,8 +438,42 @@ export const WorkspaceEdit = fp.flow(withRouteConfigData(), withCurrentWorkspace
         cloneUserRole: false,
         loading: false,
         showUnderservedPopulationDetails: false,
-        showStigmatizationDetails: false
+        showStigmatizationDetails: false,
+        diseaseList: [],
+        diseaseHover: [],
+        diseaseEntryState: 0,
       };
+      this.searchTermChangedEvent = fp.debounce(300, this.diseaseSearch);
+      this.DS_START = 1;
+      this.DS_ACTIVE = 2;
+      this.DS_SUGGEST = 4;
+      this.DS_HOVER = 8;
+    }
+
+    get showSearchResults(): boolean {
+      const state = this.state.diseaseEntryState;
+      return state === this.DS_SUGGEST || state === this.DS_HOVER;
+    }
+
+    diseaseSearch(value: string): void {
+      this.setState({diseaseList: []});
+      const searchTerm = value.trim();
+      if (!searchTerm) {
+        return;
+      }
+      const baseurl = serverConfigStore.getValue().firecloudURL;
+      const url = baseurl + '/duos/autocomplete/' + searchTerm;
```

Comment:
nit - just call the above line in this line rather than defining it in a const? more concise since we don't use `baseUrl` again